serviceerror/resource_exhausted.go

```diff
@@ -26,6 +27,14 @@ func NewResourceExhausted(cause enumspb.ResourceExhaustedCause, message string)
 	}
 }
 
+// NewResourceExhaustedf returns new ResourceExhausted error with formatted message.
+func NewResourceExhaustedf(cause enumspb.ResourceExhaustedCause, format string, args ...interface{}) error {
```

Comment:
should we have constructors that take Scope also? we always set it everywhere now, right? I feel like we'll never get to use this one because we need a Scope